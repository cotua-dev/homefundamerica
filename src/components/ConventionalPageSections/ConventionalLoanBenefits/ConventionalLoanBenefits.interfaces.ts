export interface FetchConventionalLoanBenefitsImageQuery {
    conventionalLoanBenefitsImage: {
        publicURL: string;
    };
}
