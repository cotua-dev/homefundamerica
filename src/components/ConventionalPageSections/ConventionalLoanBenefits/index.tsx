import React, {
    FunctionComponent,
    ReactElement,
} from 'react';
import {
    useStaticQuery,
    graphql,
} from 'gatsby';
import { FetchConventionalLoanBenefitsImageQuery } from './ConventionalLoanBenefits.interfaces';
import { moveToStepper } from '../../../utils/moveToStepper';
import { conventionalLoanBenefitsSection } from './ConventionalLoanBenefits.module.scss';
import * as sharedStyles from '../../sharedStyles/shared.module.scss';

export const ConventionalLoanBenefits: FunctionComponent = (): ReactElement => {
    const queryResult = useStaticQuery<FetchConventionalLoanBenefitsImageQuery>(graphql`
        query {
            conventionalLoanBenefitsImage: file(name: {eq: "AdobeStock_261487920"}, extension: {eq: "jpg"}) {
                publicURL
            }
        }
    `);

    return (
        <section id={conventionalLoanBenefitsSection} className={sharedStyles.loanNegative}>
            <div className={sharedStyles.imageWrapper}>
                <img
                    src={queryResult?.conventionalLoanBenefitsImage.publicURL}
                    alt="Conventional loan benefits image"
                />
            </div>
            <div className={sharedStyles.contentWrapper}>
                <h2>{`Conventional Loan Benefits`}</h2>
                <p>
                    {`
                        A Conventional Loan can be an option if your mortgage amount
                        doesnâ€™t exceed the conventional loan limit, $510,400 for most
                        U.S. counties.
                    `}
                </p>
                <ul>
                    <li>
                        {`One of the lowest fixed interest rates available`}
                    </li>
                    <li>
                        {`The flexibility of either a fixed-rate or an adjustable-rate mortgage (ARM)`}
                    </li>
                </ul>
                <button
                    className={sharedStyles.applyButton}
                    type="button"
                    onClick={moveToStepper}
                >Apply Now</button>
            </div>
        </section>
    );
}
