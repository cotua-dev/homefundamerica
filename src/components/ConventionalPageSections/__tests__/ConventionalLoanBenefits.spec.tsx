import React from 'react';
import {
    render,
    RenderResult,
} from '@testing-library/react';
import { ConventionalLoanBenefits } from '../ConventionalLoanBenefits';

describe('ConformingLoanBenefits Component', (): void => {
    test('renders', (): void => {
        const { container }: RenderResult = render(<ConventionalLoanBenefits/>);
        expect(container.firstChild).toMatchSnapshot();
    });
});
