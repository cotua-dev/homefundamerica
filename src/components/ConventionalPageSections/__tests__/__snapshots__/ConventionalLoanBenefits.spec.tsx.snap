// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConformingLoanBenefits Component renders 1`] = `
<section
  id="conventionalLoanBenefitsSection"
>
  <div>
    <img
      alt="Conventional loan benefits image"
    />
  </div>
  <div>
    <h2>
      Conventional Loan Benefits
    </h2>
    <p>
      
                        A Conventional Loan can be an option if your mortgage amount
                        doesnâ€™t exceed the conventional loan limit, $510,400 for most
                        U.S. counties.
                    
    </p>
    <ul>
      <li>
        One of the lowest fixed interest rates available
      </li>
      <li>
        The flexibility of either a fixed-rate or an adjustable-rate mortgage (ARM)
      </li>
    </ul>
    <button
      type="button"
    >
      Apply Now
    </button>
  </div>
</section>
`;
